flink-yarn/src/test/java/org/apache/flink/yarn/YarnResourceManagerDriverTest.java

```diff
@@ -627,9 +605,12 @@ protected void validateReleaseResources(Collection<YarnWorkerNode> workerNodes)
             verifyFutureCompleted(releaseAssignedContainerFuture);
         }
 
-        ContainerStatus createTestingContainerCompletedStatus(final ContainerId containerId) {
+        ContainerStatus createTestingContainerCompletedStatus(ContainerStatus containerStatus) {
             return new TestingContainerStatus(
-                    containerId, ContainerState.COMPLETE, "Test exit", -1);
+                    containerStatus.getContainerId(),
+                    containerStatus.getState(),
+                    containerStatus.getDiagnostics(),
+                    containerStatus.getExitStatus());
         }
```

Comment:
I don't really understand this change. Can't we keep this utility method to create a `ContainerStatus` that is `ContainerState.COMPLETE`?

Reply:
Yes, we can. I think maybe this would cause much duplicate code, so I have done some optimization.

Reply:
Maybe we can just keep `createTestingContainerCompletedStatus(final ContainerId containerId)` to create a `COMPLETED` container status with `ContainerExitStatus.SUCCESS`. For other `ContainerStatus` we can use the factory method `ContainerStatusPBImpl.newInstance`.